### DISCUSSION TITLE
Browse Unstaged/Staged changes and commit content fullscreen

### DISCUSSION THREAD

**Discussion by peauc** [jesseduffield/lazygit#3159] [Q&A]
Hello !

I love my lazygit config but there is one things that genuinely bugs me:
I cannot seem to browse changes from fullscreen. 

In the attached screenshot I would like to be able to fullscreen this panel to browse changes.
I know about the staging view by pressing `<return>` but it is not fullscreen.

Thanks a lot,
![CleanShot 2023-12-12 at 10 44 15@2x](https://github.com/jesseduffield/lazygit/assets/15132860/e7e98099-f946-41e1-a2e5-7cd1f239aa4c)

**Comment by mark2185** [comment]
Try switching to different screen modes while you're in the `Files` pane and then pressing `Enter` on a file you wish to see in more detail.

![image](https://github.com/jesseduffield/lazygit/assets/8365241/867588af-53e9-4d8e-ad9a-0872be6ab245)

**Reply by peauc** [reply]
Hello thanks for your answer, i've attached a video where I use the next screen mode.

Pressing `Enter` on a file will put me in staging mode, it's not what I want because it will split the vertical space to make space for the staging area and thus not being fullscreen and my pager is not being used.

**Reply by peauc** [reply]
https://github.com/jesseduffield/lazygit/assets/15132860/40489951-96e3-48fd-b802-26286702d73e

**Reply by mark2185** [reply]
Ah, the requirement is to use your custom `pager`.

Unfortunately that is currently not possible, the only thing that comes to mind is that you could write a custom command that opens up the currently selected file/folder in the terminal.

Try seeing what happens when running `:git diff` to see if that's viable for you.

**Reply by peauc** [reply]
Alright thanks for confirming what I thought. 
One requirement is also not to split the screen in two vertically to show stashed changes, as I only want to browse the commit's content.
I'll look into contributing for this feature. Thanks :)

**Reply by mark2185** [reply]
Feel free to come to the [discord](https://discord.gg/ehwFt2t4wt) server where we can discuss it further!

**Reply by ben-ralston** [reply]
Hey! Just curious if you found any workarounds/build for this issue?

I frequently find myself in the same situation, wishing I could stage changes in a large pane. (I also mainly use lazygit embedded in neovim, so I'm starting from slightly reduced space.)

**Comment by leerob** [comment]
I'd also love to see this 😄
